parts/kubernetesmastercustomscript.sh

```diff
@@ -103,32 +109,75 @@ function setDockerOpts () {
     sed -i "s#^DOCKER_OPTS=.*#DOCKER_OPTS=${1}#" /etc/default/kubelet
 }
 
-function configNetworkPolicy() {
+function configAzureNetworkPolicy() {
+	CNI_VERSION=v0.4.0
+	AZURE_PLUGIN_VERSION=v0.7
+
+    # Create network config file.
+    CNI_CONFIG_DIR=/etc/cni/net.d
+    NET_CONFIG_FILE=$CNI_CONFIG_DIR/10-azure.conf
+    mkdir -p $CNI_CONFIG_DIR
+
+    cat <<- EOF > $NET_CONFIG_FILE
+	{
+	  "cniVersion": "0.2.0",
+	  "name": "azure",
+	  "type": "azure-vnet",
+	  "master": "eth0",
+	  "bridge": "azure0",
+	  "ipam": {
+	    "type": "azure-vnet-ipam"
+	  }
+	}
+	EOF
+
+    chown -R root:root $CNI_CONFIG_DIR
+    chmod 755 $CNI_CONFIG_DIR
+    chmod 644 $NET_CONFIG_FILE
+
+    # Download Azure VNET CNI plugins.
+    CNI_BIN_DIR=/opt/cni/bin
+    mkdir -p $CNI_BIN_DIR
+    downloadUrl https://github.com/Azure/azure-container-networking/releases/download/$AZURE_PLUGIN_VERSION/azure-cni-linux-amd64-$AZURE_PLUGIN_VERSION.tgz | tar -xz -C $CNI_BIN_DIR
+    downloadUrl https://github.com/containernetworking/cni/releases/download/$CNI_VERSION/cni-amd64-$CNI_VERSION.tgz | tar -xz -C $CNI_BIN_DIR ./loopback
+    chown -R root:root $CNI_BIN_DIR
+    chmod -R 755 $CNI_BIN_DIR
+
+    # Dump ebtables rules.
+    /sbin/ebtables -t nat --list
+
+    # Enable CNI.
+    setNetworkPlugin cni
+    setDockerOpts " --volume=/etc/cni/:/etc/cni:ro --volume=/opt/cni/:/opt/cni:ro"
+}
+
+function configCalicoNetworkPolicy() {
     if [[ ! -z "${APISERVER_PRIVATE_KEY}" ]]; then
         # on masters
         ADDONS="calico-configmap.yaml calico-daemonset.yaml"
         ADDONS_PATH=/etc/kubernetes/addons
         CALICO_URL="https://raw.githubusercontent.com/projectcalico/calico/a4ebfbad55ab1b7f10fdf3b39585471f8012e898/v2.0/getting-started/kubernetes/installation/hosted/k8s-backend-addon-manager"
-        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
-            # download calico yamls
-            for addon in ${ADDONS}; do
-                curl -o "${ADDONS_PATH}/${addon}" -sSL --retry 12 --retry-delay 10 "${CALICO_URL}/${addon}"
-            done
-        else
-            # make sure calico yaml are removed
-            for addon in ${ADDONS}; do
-                rm -f "${ADDONS_PATH}/${addon}"
-            done
-        fi
+
+        # download calico yamls
+        for addon in ${ADDONS}; do
+            curl -o "${ADDONS_PATH}/${addon}" -sSL --retry 12 --retry-delay 10 "${CALICO_URL}/${addon}"
+        done
     else
         # on agents
-        if [[ "${NETWORK_POLICY}" = "calico" ]]; then
-            setNetworkPlugin cni
-            setDockerOpts " --volume=/etc/cni/:/etc/cni:ro --volume=/opt/cni/:/opt/cni:ro"
-        else
-            setNetworkPlugin kubenet
-            setDockerOpts ""
-        fi
+        setNetworkPlugin cni
```

Comment:
Discussed with Anthony offline. Yes, that's only called when feature is enabled.